packages/authentication/src/providers/auth-action.provider.ts

```diff
@@ -47,12 +45,9 @@ export class AuthenticateActionProvider implements Provider<AuthenticateFn> {
       // The invoked operation does not require authentication.
       return undefined;
     }
-    if (!strategy.authenticate) {
-      throw new Error('invalid strategy parameter');
-    }
-    const strategyAdapter = new StrategyAdapter(strategy);
-    const user = await strategyAdapter.authenticate(request);
-    this.setCurrentUser(user);
-    return user;
+
+    const userProfile = await strategy.authenticate(request);
+    if (userProfile) this.setCurrentUser(userProfile);
```

Comment:
Do we need error handling if `userProfile` is not returned from the strategy?

Reply:
the strategy.authenticate(request) implementations will throw an exception when : userService.verifyCredentials(user creds) fails with 401, or when tokenService.verifyToken(token) fails with 401.

Reply:
I will add a check for undefined userProfile.